test/e2e/proxy_test.go

```diff
@@ -1084,3 +1087,171 @@ func testProxyServiceLifeCycle(ipFamily *corev1.IPFamily, ingressIPs []string, d
 		}
 	}
 }
+
+// TestProxyLoadBalancerModeDSR creates a LoadBalancer Service and verifies both external and internal clients accessing
+// its LoadBalancer IPs work as expected.
+// Client IP should always be preserved regardless of whether the traffic is externally or internally originated.
+// Session affinity should take effect when it's enabled.
+func TestProxyLoadBalancerModeDSR(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNumNodesLessThan(t, 3)
+	skipIfAntreaIPAMTest(t)
+	skipIfProxyDisabled(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	skipIfProxyAllDisabled(t, data)
+	skipIfLoadBalancerModeIsNot(t, data, config.LoadBalancerModeDSR)
+	skipIfEncapModeIsNot(t, data, config.TrafficEncapModeEncap)
+
+	ingressNode := controlPlaneNodeName()
+	backendNode1 := workerNodeName(1)
+	backendNode2 := workerNodeName(2)
+
+	internalClient := "internal-client"
+	if err := NewPodBuilder(internalClient, data.testNamespace, toolboxImage).OnNode(ingressNode).WithCommand([]string{"sleep", "infinity"}).Create(data); err != nil {
```

Comment:
just an FYI that using `sleep infinity` should no longer be needed with version 1.1 of the toolbox image. I am including the K8s `pause` binary in the image, and using it as the default command: https://github.com/antrea-io/image-utils/blob/ea069f24f9b424a933f12a12ac6fa264eaa663ab/images/toolbox/Dockerfile#L52-L61